java/ql/test/library-tests/frameworks/JaxWs/JakartaRsFlow.java

```diff
@@ -0,0 +1,410 @@
+    sink(ResponseBuilderSource.taint().status(400)); // $hasValueFlow
+    sink(ResponseBuilderSource.taint().tag(new EntityTag(""))); // $hasValueFlow
+    sink(ResponseBuilderSource.taint().tag("")); // $hasValueFlow
+    sink(ResponseBuilderSource.taint().type("")); // $hasValueFlow
+    sink(ResponseBuilderSource.taint().variant(new Variant(new MediaType(), "", ""))); // $hasValueFlow
+    sink(ResponseBuilderSource.taint().variants(list)); // $hasValueFlow
+    sink(ResponseBuilderSource.taint().variants()); // $hasValueFlow
+  }
+
+  void testHttpHeaders(HttpHeaders h) {
+    sink(taint(h).getAcceptableLanguages()); // $hasTaintFlow
+    sink(taint(h).getAcceptableMediaTypes()); // $hasTaintFlow
+    sink(taint(h).getCookies()); // $hasTaintFlow
+    sink(taint(h).getHeaderString("")); // $hasTaintFlow
+    sink(taint(h).getLanguage()); // $hasTaintFlow
+    sink(taint(h).getMediaType()); // $hasTaintFlow
+    sink(taint(h).getRequestHeader("")); // $hasTaintFlow
+    sink(taint(h).getRequestHeaders()); // $hasTaintFlow
+  }
+
+  void testMultivaluedMapAdd(MultivaluedMap<String, String> mm1, MultivaluedMap<String, String> mm2) {
+    mm1.add(taint(), "value");
+    sink(mm1.keySet().iterator().next()); // $hasValueFlow
+    mm2.add("key", taint());
+    sink(mm2.get("key").get(0)); // $hasValueFlow
+  }
+
+  void testMultivaluedMapAddAll(MultivaluedMap<String, String> mm1, MultivaluedMap<String, String> mm2, MultivaluedMap<String, String> mm3) {
+    mm1.addAll(taint(), "a", "b");
+    sink(mm1.keySet().iterator().next()); // $hasValueFlow
+    List<String> l = new ArrayList<String>();
+    l.add(taint());
+    mm2.addAll("key", l);
+    sink(mm2.get("key").get(0)); // $hasValueFlow
+    mm3.addAll("key", "a", taint());
+    sink(mm3.get("key").get(0)); // $hasValueFlow
+  }
+
+  void testMultivaluedMapAddFirst(MultivaluedMap<String, String> mm1, MultivaluedMap<String, String> mm2) {
+    mm1.addFirst(taint(), "value");
+    sink(mm1.keySet().iterator().next()); // $hasValueFlow
+    mm2.addFirst("key", taint());
+    sink(mm2.get("key").get(0)); // $hasValueFlow
+    sink(mm2.getFirst("key")); // $hasValueFlow
+  }
+
+  void testMultivaluedMapputSingle(MultivaluedMap<String, String> mm1, MultivaluedMap<String, String> mm2) {
+    mm1.putSingle(taint(), "value");
+    sink(mm1.keySet().iterator().next()); // $hasValueFlow
+    mm2.putSingle("key", taint());
+    sink(mm2.get("key").get(0)); // $hasValueFlow
+  }
+
+  class MyAbstractMultivaluedMapJak<K, V> extends AbstractMultivaluedMap<K, V> {
+    public MyAbstractMultivaluedMapJak(Map<K, List<V>> map) {
+        super(map);
+    }
+  }
+
+  void testAbstractMultivaluedMap(Map<String, List<String>> map1, Map<String, List<String>> map2, List<String> list) {
+    map1.put(taint(), list);
+    AbstractMultivaluedMap<String, String> amm1 = new MyAbstractMultivaluedMapJak<String, String>(map1);
+    // sink(amm1.keySet().iterator().next()); // $hasValueFlow
```

Comment:
These two cases seem like something that could work. Why are they commented out? (Also, in general for missing results, the test expectations library supports a `MISSING` annotation).

Reply:
It will once https://github.com/github/codeql/pull/6087 lands :)

Reply:
https://github.com/github/codeql/pull/6087/commits/979393d7c217755fce0e5254d05309eda7df3a1e in particular

Reply:
Indeed they should work, and they do in isolation, but there seems to be a bug somewhere (possibly the dataflow library) which means that if these cases and the equivalent cases in `JaxRsFlow.java` are all present then they all fail. Investigating that bug shouldn't delay this PR going in. I have reinstated them with `MISSING:` and `SPURIOUS:` to make the tests pass.